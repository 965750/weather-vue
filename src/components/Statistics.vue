<template>
    <div class="mt-10">

    </div>
</template>

<script>
export default {
  name: 'Statistics',
  props: {
    weather: {
      required: true,
      type: Object,
    },
  },
};
</script>
