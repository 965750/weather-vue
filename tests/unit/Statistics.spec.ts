import {
  shallowMount,
  createLocalVue,
} from '@vue/test-utils';
import Statistics from '@/components/Statistics.vue';
import filters from '@/myFilters';

const localVue = createLocalVue();

filters.forEach((filter) => {
  localVue.filter(filter.name, filter.handler);
});

const initialProps = {
  weather: {
    mostCommon: 12,
    minTemp: 11,
    maxTemp: 15,
    clouds: 10,
    humidity: 22,
    temp: 13,
    wind: 5,
  },
};

describe('Statistics.vue', () => {
  const createWrapper = (props = initialProps) => shallowMount(Statistics, {
    localVue,
    propsData: {
      ...props,
    },
  });

  it('should match initial snapshot', () => {
    const wrapper = createWrapper();

    expect(wrapper.element).toMatchSnapshot();
  });
});
