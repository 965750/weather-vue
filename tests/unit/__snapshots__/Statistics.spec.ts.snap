// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistics.vue should match initial snapshot 1`] = `
<div
  class="d-flex justify-center flex-wrap"
>
  <singleattribute-stub
    icon="temperature-low"
    label="min. temp."
    value="-262 	℃"
  />
  <singleattribute-stub
    icon="temperature-high"
    label="max. temp."
    value="-258 	℃"
  />
  <singleattribute-stub
    icon="thermometer-half"
    label="mean temp."
    value="-260 	℃"
  />
  <singleattribute-stub
    icon="thermometer-quarter"
    label="mode temp."
    value="-261 	℃"
  />
  <singleattribute-stub
    icon="cloud"
    label="clouds"
    value="10%"
  />
  <singleattribute-stub
    icon="wind"
    label="wind"
    value="5.00 m/s"
  />
  <singleattribute-stub
    icon="tint"
    label="humidity"
    value="22%"
  />
</div>
`;
